File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set<int> numset(nums.begin(), nums.end());
        int maxlen = 0;
        for(int n : nums) {
            if (!numset.contains(n-1)) {
                int len = 0;
                while(numset.contains(n + len)) {
                    len++;
                }
                maxlen = max(maxlen, len);
            }
        }
        return maxlen;
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        hm = {}
        for n in nums:
            if n in hm:
                hm[n] += 1
            else :
                hm[n] = 1
        
        for k,v in hm.items():
            if v > 1:
                return True
        return False